mlinspect/backends/_iter_creation.py

```diff
@@ -111,20 +112,21 @@ def iter_input_annotation_output_join(inspection_count, x_data, x_annotations, y
 
     x_before_with_annotations = pandas.concat([x_data.reset_index(drop=True), x_annotations], axis=1)
     y_before_with_annotations = pandas.concat([y_data.reset_index(drop=True), y_annotations], axis=1)
-    df_x_output = pandas.merge(x_before_with_annotations, output, left_on="mlinspect_index_x",
-                               right_on="mlinspect_index_x", suffixes=["_x", "_output"])
-    df_x_output_y = pandas.merge(df_x_output, y_before_with_annotations, left_on="mlinspect_index_y",
-                                 right_on="mlinspect_index_y", suffixes=["_x_output", "_y_output"])
+    df_x_output = output.merge(x_before_with_annotations, on="mlinspect_index_x", suffixes=["_output", "_x"])
+    df_x_output_y = df_x_output.merge(y_before_with_annotations, on="mlinspect_index_y",
+                                      suffixes=["_x_output", "_y_output"])
 
-    column_index_x_end = len(x_data.columns)
+    df_x_output_y = df_x_output_y.drop(['mlinspect_index_y', 'mlinspect_index_x'], axis=1)
 
-    column_index_output_start = column_index_x_end + inspection_count
-    column_index_y_start = column_index_output_start + len(output.columns) - 2
-    column_index_y_end = column_index_y_start + len(y_data.columns) - 1
+    column_index_output_start = 0
+    column_index_output_end = len(output.columns) - 2
```

Comment:
Would it be possible to add a comment here that explains the -2 constants and the -1 constants below? If I remember correctly they are required because of the `mlinspect_index_x` and `mlinspect_index_y` columns that are still present in `output`, `x_data`, and `y_data`.

Reply:
Done in 2120ce40b21934e11bf7537437d1e51449b1db2a